1.19.2 OptiFabric collision with JEI

Crash log:
```
[16:28:00] [main/INFO]: Loading Minecraft 1.19.2 with Fabric Loader 0.14.9
[16:28:00] [main/INFO]: Loading 86 mods:
	- appleskin 2.4.1+mc1.19
	- architectury 6.2.43
	- autofish 0.9.5-SNAPSHOT
	- badpackets 0.2.0
	- bookshelf 16.1.5
	- cloth-basic-math 0.6.1 via cloth-config
	- cloth-config 8.0.75
	- craftpresence 1.8.11
	- enchdesc 13.0.1
	- fabric-api 0.60.0+1.19.2
	- fabric-api-base 0.4.11+e62f51a390 via fabric-api
	- fabric-api-lookup-api-v1 1.6.9+9ff28f4090 via fabric-api
	- fabric-biome-api-v1 9.0.17+aeb40ebe90 via fabric-api
	- fabric-blockrenderlayer-v1 1.1.20+aeb40ebe90 via fabric-api
	- fabric-client-tags-api-v1 1.0.1+b35fea8390 via fabric-api
	- fabric-command-api-v1 1.2.11+f71b366f90 via fabric-api
	- fabric-command-api-v2 2.1.7+0c17ea9690 via fabric-api
	- fabric-commands-v0 0.2.28+df3654b390 via fabric-api
	- fabric-containers-v0 0.1.34+df3654b390 via fabric-api
	- fabric-content-registries-v0 3.2.3+aeb40ebe90 via fabric-api
	- fabric-convention-tags-v1 1.1.1+7cd20a1490 via fabric-api
	- fabric-crash-report-info-v1 0.2.5+aeb40ebe90 via fabric-api
	- fabric-data-generation-api-v1 5.1.8+a680b9b490 via fabric-api
	- fabric-dimensions-v1 2.1.30+aeb40ebe90 via fabric-api
	- fabric-entity-events-v1 1.4.18+9ff28f4090 via fabric-api
	- fabric-events-interaction-v0 0.4.28+aeb40ebe90 via fabric-api
	- fabric-events-lifecycle-v0 0.2.28+df3654b390 via fabric-api
	- fabric-game-rule-api-v1 1.0.21+aeb40ebe90 via fabric-api
	- fabric-item-api-v1 1.5.7+35a03c4390 via fabric-api
	- fabric-item-groups-v0 0.3.28+aeb40ebe90 via fabric-api
	- fabric-key-binding-api-v1 1.0.20+aeb40ebe90 via fabric-api
	- fabric-keybindings-v0 0.2.18+df3654b390 via fabric-api
	- fabric-language-kotlin 1.8.3+kotlin.1.7.10
	- fabric-lifecycle-events-v1 2.1.2+aeb40ebe90 via fabric-api
	- fabric-loot-api-v2 1.1.3+83a8659290 via fabric-api
	- fabric-loot-tables-v1 1.1.6+9e7660c690 via fabric-api
	- fabric-message-api-v1 5.0.3+176380a290 via fabric-api
	- fabric-mining-level-api-v1 2.1.14+33fbc73890 via fabric-api
	- fabric-models-v0 0.3.17+aeb40ebe90 via fabric-api
	- fabric-networking-api-v1 1.2.4+5eb68ef290 via fabric-api
	- fabric-networking-v0 0.3.21+df3654b390 via fabric-api
	- fabric-object-builder-api-v1 4.0.11+aeb40ebe90 via fabric-api
	- fabric-particles-v1 1.0.10+aeb40ebe90 via fabric-api
	- fabric-registry-sync-v0 0.9.24+aeb40ebe90 via fabric-api
	- fabric-renderer-api-v1 1.0.10+aeb40ebe90 via fabric-api
	- fabric-renderer-indigo 0.6.12+aeb40ebe90 via fabric-api
	- fabric-renderer-registries-v1 3.2.20+df3654b390 via fabric-api
	- fabric-rendering-data-attachment-v1 0.3.14+aeb40ebe90 via fabric-api
	- fabric-rendering-fluids-v1 3.0.7+aeb40ebe90 via fabric-api
	- fabric-rendering-v0 1.1.22+df3654b390 via fabric-api
	- fabric-rendering-v1 1.10.16+aeb40ebe90 via fabric-api
	- fabric-resource-conditions-api-v1 2.0.11+e62f51a390 via fabric-api
	- fabric-resource-loader-v0 0.6.1+aeb40ebe90 via craftpresence
	- fabric-screen-api-v1 1.0.26+f05effaf90 via fabric-api
	- fabric-screen-handler-api-v1 1.2.13+9ff28f4090 via fabric-api
	- fabric-textures-v0 1.0.20+aeb40ebe90 via fabric-api
	- fabric-transfer-api-v1 2.0.11+83a8659290 via fabric-api
	- fabric-transitive-access-wideners-v1 1.3.0+42d99c3290 via fabric-api
	- fabricloader 0.14.9
	- fiber 0.23.0-2 via neat
	- imblocker 1.0.19
	- inventoryhud 3.4.2
	- inventoryprofilesnext 1.7.1
	- java 17
	- jei 11.3.0.262
	- lightoverlay 7.0.0
	- minecraft 1.19.2
	- mm 2.3 via optifabric
	- modmenu 4.0.6
	- neat 1.19-32-FABRIC
	- optifabric 1.13.16
	- org_jetbrains_kotlin_kotlin-reflect 1.7.10 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib 1.7.10 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib-jdk7 1.7.10 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib-jdk8 1.7.10 via fabric-language-kotlin
	- org_jetbrains_kotlinx_atomicfu-jvm 0.18.3 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm 1.6.4 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-coroutines-jdk8 1.6.4 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-datetime-jvm 0.4.0 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm 1.4.0 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-core-jvm 1.4.0 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-json-jvm 1.4.0 via fabric-language-kotlin
	- smoothboot 1.19-1.7.1
	- wthit 5.12.0
	- xaerominimap 22.14.0
	- xaeroworldmap 1.27.0
[16:28:03] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/F:/MultiMC/libraries/net/fabricmc/sponge-mixin/0.11.4+mixin.0.8.5/sponge-mixin-0.11.4+mixin.0.8.5.jar Service=Knot/Fabric Env=CLIENT
[16:28:03] [main/INFO]: Compatibility level set to JAVA_16
[16:28:03] [main/INFO]: Compatibility level set to JAVA_17
[16:28:04] [main/INFO]: Smooth Boot config initialized
[16:28:04] [main/WARN]: Configuration conflict: there is more than one oshi.properties file on the classpath
[16:28:04] [main/WARN]: Configuration conflict: there is more than one oshi.architecture.properties file on the classpath
[16:28:04] [main/WARN]: @Final field field_22786:Ljava/util/List; in mixins.ipnext.json:MixinScreen from mod inventoryprofilesnext should be final
[16:28:04] [main/WARN]: @Final field field_33815:Ljava/util/List; in mixins.ipnext.json:MixinScreen from mod inventoryprofilesnext should be final
[16:28:05] [main/ERROR]: Minecraft has crashed!
net.fabricmc.loader.impl.FormattedException: java.lang.ExceptionInInitializerError
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:463) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.9.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:210) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:245) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
Caused by: java.lang.ExceptionInInitializerError
	at net.minecraft.class_128.method_568(class_128.java:175) ~[client-intermediary.jar:?]
	at net.minecraft.class_128.method_24305(class_128.java:345) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:159) ~[minecraft-1.19.2-client.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[minecraft-1.19.2-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.9.jar:?]
	... 10 more
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_309 failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:419) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.9.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.Class.getDeclaredFields0(Native Method) ~[?:?]
	at java.lang.Class.privateGetDeclaredFields(Class.java:3297) ~[?:?]
	at java.lang.Class.getDeclaredFields(Class.java:2371) ~[?:?]
	at net.optifine.reflect.FieldLocatorTypes.<init>(FieldLocatorTypes.java:25) ~[Optifine-mapped.jar:?]
	at net.optifine.reflect.Reflector.<clinit>(Reflector.java:501) ~[Optifine-mapped.jar:?]
	at net.minecraft.class_128.method_568(class_128.java:175) ~[client-intermediary.jar:?]
	at net.minecraft.class_128.method_24305(class_128.java:345) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:159) ~[minecraft-1.19.2-client.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[minecraft-1.19.2-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.9.jar:?]
	... 10 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:414) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.9.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.Class.getDeclaredFields0(Native Method) ~[?:?]
	at java.lang.Class.privateGetDeclaredFields(Class.java:3297) ~[?:?]
	at java.lang.Class.getDeclaredFields(Class.java:2371) ~[?:?]
	at net.optifine.reflect.FieldLocatorTypes.<init>(FieldLocatorTypes.java:25) ~[Optifine-mapped.jar:?]
	at net.optifine.reflect.Reflector.<clinit>(Reflector.java:501) ~[Optifine-mapped.jar:?]
	at net.minecraft.class_128.method_568(class_128.java:175) ~[client-intermediary.jar:?]
	at net.minecraft.class_128.method_24305(class_128.java:345) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:159) ~[minecraft-1.19.2-client.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[minecraft-1.19.2-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.9.jar:?]
	... 10 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Callback method beforeCharTypedEvent(Lnet/minecraft/class_364;IILorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V in jei.mixins.json:KeyboardHandlerMixin from mod jei failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap jei-1.19.2-fabric-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1385) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1052) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:414) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.9.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.Class.getDeclaredFields0(Native Method) ~[?:?]
	at java.lang.Class.privateGetDeclaredFields(Class.java:3297) ~[?:?]
	at java.lang.Class.getDeclaredFields(Class.java:2371) ~[?:?]
	at net.optifine.reflect.FieldLocatorTypes.<init>(FieldLocatorTypes.java:25) ~[Optifine-mapped.jar:?]
	at net.optifine.reflect.Reflector.<clinit>(Reflector.java:501) ~[Optifine-mapped.jar:?]
	at net.minecraft.class_128.method_568(class_128.java:175) ~[client-intermediary.jar:?]
	at net.minecraft.class_128.method_24305(class_128.java:345) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:159) ~[minecraft-1.19.2-client.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[minecraft-1.19.2-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.9.jar:?]
	... 10 more

```

--- a/src/main/java/me/modmuss50/optifabric/compat/architectury/mixin/GameRendererNewerererMixin.java
+++ b/src/main/java/me/modmuss50/optifabric/compat/architectury/mixin/GameRendererNewerererMixin.java
@@ -1,0 +1,47 @@
+package me.modmuss50.optifabric.compat.architectury.mixin;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.At.Shift;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.minecraft.client.render.GameRenderer;
+import net.minecraft.client.util.Window;
+import net.minecraft.client.util.math.MatrixStack;
+import org.joml.Matrix4f;
+
+import me.modmuss50.optifabric.compat.InterceptingMixin;
+import me.modmuss50.optifabric.compat.PlacatingSurrogate;
+import me.modmuss50.optifabric.compat.Shim;
+
+@Mixin(GameRenderer.class)
+@InterceptingMixin("dev/architectury/mixin/fabric/client/MixinGameRenderer")
+abstract class GameRendererNewerererMixin {
+    @Shim
+    private native void renderScreenPre(float tickDelta, long startTime, boolean tick, CallbackInfo ci, int mouseX, int mouseY, Window window, Matrix4f matrix, MatrixStack matrices, MatrixStack matrices2);
+
+    @PlacatingSurrogate
+    private void renderScreenPre(float tickDelta, long startTime, boolean tick, CallbackInfo ci, int mouseX, int mouseY, Window window, float guiFarPlane, Matrix4f matrix, MatrixStack matrices) {
+    }
+
+    @Inject(method = "render(FJZ)V", locals = LocalCapture.CAPTURE_FAILHARD, cancellable = true,
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screen/Screen;renderWithTooltip(Lnet/minecraft/client/util/math/MatrixStack;IIF)V", ordinal = 0))
+    private void renderScreenPre(float tickDelta, long startTime, boolean tick, CallbackInfo ci, int mouseX, int mouseY, Window window, float guiFarPlane, Matrix4f matrix, MatrixStack matrices, MatrixStack matrices2) {
+        renderScreenPre(tickDelta, startTime, tick, ci, mouseX, mouseY, window, matrix, matrices, matrices2);
+    }
+
+    @Shim
+    private native void renderScreenPost(float tickDelta, long startTime, boolean tick, CallbackInfo ci, int mouseX, int mouseY, Window window, Matrix4f matrix, MatrixStack matrices, MatrixStack matrices2);
+
+    @PlacatingSurrogate
+    private void renderScreenPost(float tickDelta, long startTime, boolean tick, CallbackInfo ci, int mouseX, int mouseY, Window window, float guiFarPlane, Matrix4f matrix, MatrixStack matrices) {
+    }
+
+    @Inject(method = "render(FJZ)V", locals = LocalCapture.CAPTURE_FAILHARD,
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screen/Screen;renderWithTooltip(Lnet/minecraft/client/util/math/MatrixStack;IIF)V", shift = Shift.AFTER, ordinal = 0))
+    private void renderScreenPost(float tickDelta, long startTime, boolean tick, CallbackInfo ci, int mouseX, int mouseY, Window window, float guiFarPlane, Matrix4f matrix, MatrixStack matrices, MatrixStack matrices2) {
+        renderScreenPost(tickDelta, startTime, tick, ci, mouseX, mouseY, window, matrix, matrices, matrices2);
+    }
+}

--- a/src/main/java/me/modmuss50/optifabric/mod/OptifabricSetup.java
+++ b/src/main/java/me/modmuss50/optifabric/mod/OptifabricSetup.java
@@ -397,7 +397,9 @@ public class OptifabricSetup implements Runnable {
 			Mixins.addConfiguration("optifabric.compat.images.mixins.json");
 		}
 
-		if (isPresent("architectury", ">=3.7")) {
+		if (isPresent("architectury", ">=7.0.52")) {
+			Mixins.addConfiguration("optifabric.compat.architectury-AB.newererer-mixins.json");
+		} else if (isPresent("architectury", ">=3.7")) {
 			Mixins.addConfiguration("optifabric.compat.architectury-AB.newerer-mixins.json");
 		} else if (isPresent("architectury", ">=2.0")) {
 			assert isPresent("minecraft", ">=1.17-beta.1");

--- a/src/main/java/me/modmuss50/optifabric/patcher/fixes/KeyboardFix.java
+++ b/src/main/java/me/modmuss50/optifabric/patcher/fixes/KeyboardFix.java
@@ -1,5 +1,6 @@
 package me.modmuss50.optifabric.patcher.fixes;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.MoreCollectors;
 
 import org.apache.commons.lang3.Validate;
@@ -14,13 +15,25 @@ import org.objectweb.asm.tree.VarInsnNode;
 
 import me.modmuss50.optifabric.util.RemappingUtils;
 
+import java.util.List;
+
 public class KeyboardFix implements ClassFixer {
 	//net/minecraft/client/Keyboard.onKey(JIIII)V
 	private final String onKeyName = RemappingUtils.getMethodName("class_309", "method_1466", "(JIIII)V");
+	private final String screenClass = RemappingUtils.getClassName("class_437");
+	private final List<String> onKeyLambdasNames = ImmutableList.of(
+			RemappingUtils.getMethodName("class_309", "method_1458", "(L" + screenClass + ";II)V"),
+			RemappingUtils.getMethodName("class_309", "method_1473", "(L" + screenClass + ";CI)V"));
 
 	@Override
 	public void fix(ClassNode optifine, ClassNode minecraft) {
 		Validate.notNull(onKeyName, "onKeyName null");
+
+		//Needed for JEI, might as well remove the useless forge checks optifine does
+		optifine.methods.removeIf(methodNode -> onKeyLambdasNames.contains(methodNode.name));
+		minecraft.methods.stream()
+				.filter(methodNode -> onKeyLambdasNames.contains(methodNode.name))
+				.forEach(methodNode -> optifine.methods.add(methodNode));
 
 		//Remove the old "broken" method
 		optifine.methods.removeIf(methodNode -> methodNode.name.equals(onKeyName));

--- a/src/main/resources/optifabric.compat.architectury-AB.newererer-mixins.json
+++ b/src/main/resources/optifabric.compat.architectury-AB.newererer-mixins.json
@@ -1,0 +1,8 @@
+{
+    "parent": "optifabric.mixins.json",
+    "package": "me.modmuss50.optifabric.compat.architectury.mixin",
+    "plugin": "me.modmuss50.optifabric.compat.InterceptingMixinPlugin",
+    "mixins": [
+        "GameRendererNewerererMixin"
+    ]
+}

